src/main/java/seedu/address/model/article/UniqueArticleList.java

```diff
@@ -165,4 +170,45 @@ private boolean articlesAreUnique(List<Article> articles) {
         }
         return true;
     }
+
+    /**
+     * Looks up the article with the given articleID in the article book.
+     * Returns a list of persons who are related to the article.
+     */
+    public List<Person> lookupArticle(String articleID, AddressBook addressBook) throws CommandException {
+        requireAllNonNull(articleID, addressBook);
+        Article article = this.lookupArticle(articleID);
+        List<Person> persons = new ArrayList<>();
+
+        for (Author authors : article.getAuthors()) {
+            for (Person person : addressBook.getPersonList()) {
+                if (person.getName().fullName.equals(authors.authorName)) {
+                    persons.add(person);
+                }
+            }
+        }
+        for (Source sources : article.getSources()) {
+            for (Person person : addressBook.getPersonList()) {
+                if (person.getName().fullName.equals(sources.sourceName)) {
+                    persons.add(person);
+                }
+            }
+        }
+        article.setPersonList(persons);
+        return article.getPersonList();
+    }
+
+    /**
+     * Looks up the article with the given articleID in the article book.
+     * Returns the article if it exists, otherwise returns null.
```

Comment:
By the current implementation, it will only throw a `CommandException` and never return a null value, so this comment is erronous.